extensions/machine-learning-services/src/modelManagement/registeredModelService.ts

```diff
@@ -33,20 +35,57 @@ export class RegisteredModelService {
 			let result = await this.runRegisteredModelsListQuery(connection);
 			if (result && result.rows && result.rows.length > 0) {
 				result.rows.forEach(row => {
-					list.push({
-						id: +row[0].displayValue,
-						name: row[1].displayValue
-					});
+					list.push(this.loadModelData(row));
 				});
 			}
 		}
 		return list;
 	}
 
-	public async registerLocalModel(filePath: string) {
+	private loadModelData(row: azdata.DbCellValue[]): RegisteredModel {
+		return {
```

Comment:
Could this return index out of bounds or attribute errors?

Reply:
no because the name of columns are hard coded so they can be less than expected

---
In reply to: [386572418](https://github.com/microsoft/azuredatastudio/pull/9377#discussion_r386572418) [](ancestors = 386572418)